time_execution/timed.py

```diff
@@ -0,0 +1,91 @@
+from __future__ import annotations
+
+from contextlib import AbstractContextManager
+from socket import gethostname
+from timeit import default_timer
+from types import TracebackType
+from typing import Any, Callable, Dict, List, Optional, Tuple, Type
+
+from time_execution import settings, write_metric
+
+SHORT_HOSTNAME = gethostname()
+
+
+class Timed(AbstractContextManager):
+    """
+    Both the sync and async decorators require the same logic around the wrapped function.
+    This context manager encapsulates the shared behaviour to avoid duplicating the code.
+    """
+
+    __slots__ = (
```

Comment:
could maybe become a dataclass with slots=True

Reply:
I could make it a `@dataclass`, alright, but `slots` is only available since 3.10

Reply:
Ah.. then let's keep it this way for now